app/models/domain/StringFieldRegex.scala

```diff
@@ -24,6 +24,7 @@ object StringFieldRegex {
   val alphaNumericRegex: Regex           = "^[a-zA-Z0-9]*$".r
   val numericRegex: Regex                = "^[0-9]*$".r
   val alphaNumericWithSpacesRegex: Regex = "^[a-zA-Z\\s0-9]*$".r
+  val carrierEoriRegex                   = "^([a-zA-Z]{2,2}[a-zA-Z|\\d]{1,15})$".r
```

Comment:
Can perhaps be simplified to `"^([a-zA-Z]{2}[a-zA-Z0-9|]{1,15})$".r`

Reply:
Done